### ISSUE TITLE
Extend generic scattering calculator with tabs for other simulators

### ISSUE THREAD

**Issue by toqduj** [SasView/sasview#2011]
We've got a few options for additional scattering pattern simulators. Additionally, we can use simulated curves from this interface as models for SasView (which then adds polydispersity and structure factors, albeit only in a limited globular-assumed-fashion). 

So after discussion with @rozyczko, we are thinking to extend the generic scattering calculator interface to add 
1) tabs for other simulators such as the SPONGE and the McSIM from Anders H. Larsen (and maybe also others), and 
2) add a button that basically does the conversion I presented at the meeting last week, which turns a simulated dataset into a model through interpolation and extrapolation. 

![Screen Shot 2022-01-28 at 9 32 56](https://user-images.githubusercontent.com/5449929/151514104-df5013de-26f0-4f72-a1bf-209777c90ff6.png)

**Comment by toqduj** [comment]
@smk78 I agree, was going to submit a ticket for such an interface as well, but I haven't made the mock-up yet. the interface needs to guide the users through the steps necessary in a logical manner..

**Comment by andreashlarsen** [comment]
It would be great - I hope it is feasible. You can see the McSim UI at genapp: 
https://somo.chem.utk.edu/mcsim/
it is not super fancy... and the underlying code is written in python

**Comment by butlerpd** [comment]
I'm not sure I understand the need for so many tabs here.  It seems to me that for a this type of calculator there are a couple of steps and each might have different ways of achieving it. Would this not be similar to choosing which fit engine to use?

The first step is the generation of a file that encodes the location and scattering lengths (or scattering length densities depending) of points that form the scattering object... which, by the way, **does not need to be a particle**. This SasView currently does not provide many (or any?) tools for other than to read some files.  The number of file types should in fact be expanded.  It would be nice to add a tool that helps generate these in the first place.

The second step is to take those points in space and generate the expected scattering pattern, and/or, as Andreas suggests, other patterns such as the P(r) which can be calculated directly from the points.

A third step could include overlaying those results with data and/or calculating residuals etc

rather than tabs for different calculators, wouldn't a drop down as currently exists be more appropriate? currently only two methods are implemented but there are a number of others that could definitely be added such as the golden vector approach etc.  This would seem like a cleaner, less confusing interface to me. But I may be misunderstanding some of the choices envisaged here.

Which leads me to another question: should not the "generic scattering calculator," thus defined, be moved to be another perspective? Arguably it is not really "Analysis" though if compared to a data set and/or added "residual" plots it gets awful close.

**Comment by smk78** [comment]
> Which leads me to another question: should not the "generic scattering calculator," thus defined, be moved to be another perspective?

Yes.

It's arguably already grown beyond a simple tool, and the scope of the suggested enhancements would drive it further towards being another way of interpreting data which is, of course, the definition of analysis.

**Comment by andreashlarsen** [comment]
McSim inclusion has got its own issue: #2023
